src/emqx_connection.erl

```diff
@@ -446,7 +450,7 @@ handle_msg(Msg, State) ->
 -spec terminate(any(), state()) -> no_return().
 terminate(Reason, State = #state{channel = Channel, transport = Transport,
           socket = Socket}) ->
-    ?LOG(debug, "Terminated due to ~p", [Reason]),
```

Comment:
Has the print log been deleted from this line?

I would recommend keeping this DEBUG log. When we sometimes analyze our customers' logs, we often determine the point at which a device disconnects based on this line.

Reply:
tp macro in testing is a trace point, in production it’s a log